interpreter/terminal_interface/magic_commands.py

```diff
@@ -72,47 +82,131 @@ def handle_debug(self, arguments=None):
-      json_path = "messages.json"
+        json_path = "messages.json"
     if not json_path.endswith(".json"):
-      json_path += ".json"
+        json_path += ".json"
     with open(json_path, 'w') as f:
-      json.dump(self.messages, f, indent=2)
+        json.dump(self.messages, f, indent=2)
+
+    display_markdown_message(
+        f"> messages json export to {os.path.abspath(json_path)}")
 
-    display_markdown_message(f"> messages json export to {os.path.abspath(json_path)}")
 
 def handle_load_message(self, json_path):
     if json_path == "":
-      json_path = "messages.json"
+        json_path = "messages.json"
     if not json_path.endswith(".json"):
-      json_path += ".json"
+        json_path += ".json"
     with open(json_path, 'r') as f:
-      self.messages = json.load(f)
+        self.messages = json.load(f)
+
+    display_markdown_message(
+        f"> messages json loaded from {os.path.abspath(json_path)}")
+
+def handle_container_upload(self, *args):
+
+    if self.use_containers:
+        try:
+            client = docker.APIClient()
+        except:
+            error_message = (
+                "We were not able to connect to the Docker Container daemon. "
+                "Please ensure Docker is installed and running. If you have not run any code yet, "
+                "you will need to in order to start a container."
+            )
+            display_markdown_message(f"{error_message}")
+            return
+
+        for filepath in args:
+            if os.path.exists(filepath):
+                session_id = SESSION_IDS_BY_OBJECT.get(self)
+                containers = client.containers(filters={"label": f"session_id={session_id}"})
+                if containers:
+                    container_id = containers[0]['Id']
+                    # /mnt/data is default workdir for container
+                    copy_file_to_container(
+                        container_id=container_id, local_path=filepath, path_in_container=f"/mnt/data/{os.path.basename(filepath)}"
+                    )
+                    success_message = f"File [{filepath}](#) successfully uploaded to container in dir `/mnt/data`."
+                    display_markdown_message(success_message)
+                else:
+                    no_container_message = (
+                        "No container found to upload to. Please run any code to start one. "
+                        "This will be fixed in a later update."
+                    )
+                    display_markdown_message(f"**'{no_container_message}'**")
+            else:
+                file_not_found_message = f"File `{filepath}` does not exist."
+                display_markdown_message(file_not_found_message)
+    else:
+        ignore_command_message = "File uploads are only used when using containerized code execution. Ignoring command."
+        display_markdown_message(f"**{ignore_command_message}**")
+        
+def handle_container_download(self, *args):
```

Comment:
It looks like `%download` requires the path to include `/mnt/data/`, which isn't ideal because the `%upload` command just puts the file in that path, and `/mnt/data/` is the current working directory, so I think you should just be able to `%download filename.type` instead of `%download /mnt/data/filename.type`.

The `/mnt/data` directory is an implementation detail that the end user shouldn't need to care about.

![Screen Shot 2023-10-08 at 12 34 05 PM](https://github.com/KillianLucas/open-interpreter/assets/1667415/9b7a5dc4-9238-4f62-ab57-938c995c9d77)

Reply:
Also, just as a note, in a future update, it would be nice to be able to configure where the files are downloaded to by default.

Reply:
agreed. I am also working on making it so that for %upload on a GUI based system, it will open the systems file-explorer window to make it a little more intuitive.

Reply:
Also, there was an interesting feature sort of buried in the description of this #585:

If the user just supplies a file path (like maybe they dragged a file into the terminal), it would be pretty neat if it asked if I wanted to upload the file to the container.

Reply:
I'm going to resolve this one and we can focus on it as a follow up once we get this MVP merged in and don't have to fight against merge conflicts with such a large changeset.

The overall functionality here is awesome and we can work on incrementally improving the User Experience and Developer Experience.